app/src/androidTest/java/de/test/antennapod/storage/DBTasksTest.java

```diff
@@ -179,4 +184,86 @@ private void updatedFeedTest(final Feed newFeed, long feedID, List<Long> itemIDs
             lastDate = item.getPubDate();
         }
     }
+
+    @Test
+    public void testAddQueueItemsInDownload_EnqueueEnabled() throws Exception {
+        // Setup test data / environment
+        UserPreferences.setEnqueueDownloadedEpisodes(true);
+        UserPreferences.setEnqueueAtFront(false);
+
+        List<FeedItem> fis1 = createSavedFeed("Feed 1", 2).getItems();
+        List<FeedItem> fis2 = createSavedFeed("Feed 2", 3).getItems();
+
+        DBWriter.addQueueItem(context, fis1.get(0), fis2.get(0)).get();
+        // the first item fis1.get(0) is already in the queue
+        FeedItem[] itemsToDownload = new FeedItem[]{ fis1.get(0), fis1.get(1), fis2.get(2), fis2.get(1) };
+
+        // Expectations:
+        List<FeedItem> expectedEnqueued = Arrays.asList(fis1.get(1), fis2.get(2), fis2.get(1));
+        List<FeedItem> expectedQueue = new ArrayList<>();
+        expectedQueue.addAll(DBReader.getQueue());
+        expectedQueue.addAll(expectedEnqueued);
+
+        // Run actual test and assert results
+        List<? extends FeedItem> actualEnqueued =
+                DBTasks.enqueueFeedItemsToDownload(context, itemsToDownload);
+
+        assertEqualsByIds("Only items not in the queue are enqueued", expectedEnqueued, actualEnqueued);
+        assertEqualsByIds("Queue has new items appended", expectedQueue, DBReader.getQueue());
+    }
+
+    @Test
+    public void testAddQueueItemsInDownload_EnqueueDisabled() throws Exception {
+        // Setup test data / environment
+        UserPreferences.setEnqueueDownloadedEpisodes(false);
+
+        List<FeedItem> fis1 = createSavedFeed("Feed 1", 2).getItems();
+        List<FeedItem> fis2 = createSavedFeed("Feed 2", 3).getItems();
+
+        DBWriter.addQueueItem(context, fis1.get(0), fis2.get(0)).get();
+        FeedItem[] itemsToDownload = new FeedItem[]{ fis1.get(0), fis1.get(1), fis2.get(2), fis2.get(1) };
+
+        // Expectations:
+        List<FeedItem> expectedEnqueued = Collections.emptyList();
+        List<FeedItem> expectedQueue = DBReader.getQueue();
+
+        // Run actual test and assert results
+        List<? extends FeedItem> actualEnqueued =
+                DBTasks.enqueueFeedItemsToDownload(context, itemsToDownload);
+
+        assertEqualsByIds("No item is enqueued", expectedEnqueued, actualEnqueued);
+        assertEqualsByIds("Queue is unchanged", expectedQueue, DBReader.getQueue());
+    }
+
+    private void assertEqualsByIds(String msg, List<? extends FeedItem> expected, List<? extends FeedItem> actual) {
+        // assert only the IDs, so that any differences are easily to spot.
+        List<Long> expectedIds = toIds(expected);
+        List<Long> actualIds = toIds(actual);
+        assertEquals(msg, expectedIds, actualIds);
+    }
+
+    private static List<Long> toIds(List<? extends FeedItem> items) {
```

Comment:
`ItemEnqueuePositionCalculatorTest` seems to have a copy of the same method. Not sure where to put the method but I don't like code duplication.

Reply:
It could possibly be moved to main code base's `FeedItemUtil`. I just looked at it. It has a similar method that returns `long[]`  instead of collection.

Reply:
Streams are not part of the Android API <24. This could lead to crashes if it is accidentally used outside of tests.